includes/wikia/parser/templatetypes/handlers/ContextLinkTemplate.class.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+class ContextLinkTemplate {
+
+	/**
+	 * @desc check if template should be processed
+	 *
+	 * @param $templateWikitext
+	 *
+	 * @return bool
+	 */
+	public static function templateShouldBeProcessed( $templateWikitext ) {
+		global $wgEnableTemplateTypesParsing, $wgArticleAsJson;
+
+		return $wgEnableTemplateTypesParsing
+		&& $wgArticleAsJson
+		&& !empty( $templateWikitext );
+	}
+
+	/**
+	 * @desc sanitize context-link template content
+	 *
+	 * @param $wikitext
+	 *
+	 * @return string
+	 */
+	public static function handle( $wikitext ) {
+		if ( !self::containNestedTemplates( $wikitext ) ) {
```

Comment:
I'm not sure how to read this; What will happen if the context link template DOES contain nested templates?